wildwood/_split.py

```diff
@@ -433,52 +402,11 @@ def find_best_split_classifier_along_feature(
 
     is_feature_categorical = tree_context.is_categorical[feature]
 
-    # TODO: We don't need to instantiate it here
-    bins_train = np.empty(non_empty_bins_train_count, dtype=np.uint8)
-    bins_valid = np.empty(non_empty_bins_valid_count, dtype=np.uint8)
-
-    cumsum_samples_valid_left = np.zeros(n_samples_train_in_bins.size, dtype=np.uintp)
-
-    if is_feature_categorical:
-        # TODO: Something faster ? Use a placeholder for y_sum_positives in split ?
-        # Sort bins according to the proportions of labels equal to 1 in
-        # each bin. This leads to the best split partitioning for regression and binary
-        # classification
-        y_sum_positives = y_sum_in_bins[non_empty_bins_train, 1].copy()
-        y_sum_positives /= y_sum_in_bins[non_empty_bins_train, :].sum(axis=1)
-        idx_sort = np.argsort(y_sum_positives)
-        bins_train[:] = non_empty_bins_train[idx_sort]
-    else:
-        # When the feature is continuous, we simply sort the bins in ascending order
-        bins_train[:] = np.sort(non_empty_bins_train)
-        bins_valid[:] = np.sort(non_empty_bins_valid)
-        # We need to compute carefully the number of valid samples along the
-        # bins_train array (since bins_train and bins_valid might not share the same
-        # support). Let us give an example : let's say that max_bins=10 and that
-        #   n_samples_train_in_bins = [0. 3. 0. 0. 2. 1. 5. 0. 1. 0.]
-        # and that
-        #   n_samples_valid_in_bins = [2. 0. 0. 1. 1. 0. 2. 1. 1. 1.].
-        # Then we'll have
-        #    cum_samples_valid_left = [0. 2. 0. 0. 4. 4. 6. 0. 0. 0.].
-        # For any bin_train in bins_train, we want cum_samples_valid_left[bin_train] to
-        # contain the number of valid samples on the left of bin_train (including it).
-        idx_valid = 0
-        bin_test = bins_valid[idx_valid]
-        cum_valid_left = 0
-        for bin_train in bins_train:
-            while bin_test <= bin_train and idx_valid < bins_valid.size:
-                cum_valid_left += n_samples_valid_in_bins[bin_test]
-                idx_valid += 1
-                if idx_valid == bins_valid.size:
-                    break
-                bin_test = bins_valid[idx_valid]
-            cumsum_samples_valid_left[bin_train] = cum_valid_left
-
     # We go from left to right and compute the information gain proxy of all possible
     #  splits in order to find the best one. Since bin_threshold is included in the
     #  left child, we must stop before the last bin, hence the bins[:-1]
     #  (otherwise n_samples_train_right == 0 for bin_threshold=bins[-1])
-    for bin_threshold in bins_train[:-1]:
+    for bin_threshold in ordered_bins_train[:-1]:
```

Comment:
Keep this variable named bins_train since it's the same as in find_split_along_feature*